nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/launcher/connect/ConnectBroker.java

```diff
@@ -566,25 +586,44 @@ public boolean pkgPurge() throws PackageException {
      */
     public boolean pkgUninstall(List<String> packageIdsToRemove) {
         log.debug("Uninstalling: " + packageIdsToRemove);
-        for (String pkgId : packageIdsToRemove) {
+        Queue<String> remaining = new LinkedList<>(packageIdsToRemove);
+        while (!remaining.isEmpty()) {
+            String pkgId = remaining.poll();
             if (pkgUninstall(pkgId) == null) {
                 log.error("Unable to uninstall " + pkgId);
                 return false;
             }
+            if (isRestartRequired()) {
+                persistCommand(serializeUninstallCmd(remaining));
+                throw new LauncherRestartException();
+            }
         }
         return true;
     }
 
+    String serializeUninstallCmd(Collection<String> packages) {
+        if (packages.isEmpty()) {
+            return "";
+        } else {
+            return CommandInfo.CMD_UNINSTALL + " " + String.join(" ", packages);
+        }
+    }
+
+    String serializeInstallCmd(Collection<String> packages) {
```

Comment:
Not sure if this is guideline, I would prefer to use `protected` visibility rather than package protected one.